libs/category/state/src/reducers/category/category.reducer.ts

```diff
@@ -43,116 +44,135 @@ export function daffCategoryReducer<U extends DaffGenericCategory<U>, W extends
   action: DaffCategoryActions<U, W> | DaffCategoryPageActions<U, W> | DaffCategoryPageFilterActions,
 ): DaffCategoryReducerState {
   switch (action.type) {
-    case DaffCategoryActionTypes.CategoryLoadAction:
-      return {
-        ...state,
-        categoryLoading: true,
-        productsLoading: true,
-        categoryPageConfigurationState: {
-          ...state.categoryPageConfigurationState,
-          daffState: DaffState.Resolving,
-        },
-      };
-
-    // This reducer must assume the call will be successful, and immediately set the applied filters to state, because the
-    // GetCategory magento call doesn't return currently applied filters. If there is a bug where the wrong filters are somehow
-    // applied by Magento, then this will result in a bug. Until Magento returns applied filters with a category call, this is
-    // unavoidable.
     case DaffCategoryPageActionTypes.CategoryPageLoadAction:
       return {
         ...state,
         categoryLoading: true,
         productsLoading: true,
-        categoryPageConfigurationState: {
-          ...state.categoryPageConfigurationState,
+        daffState: DaffState.Resolving,
+        categoryPageMetadata: {
+          ...state.categoryPageMetadata,
           ...action.request,
```

Comment:
On second thought, I think Damien mentioned not spreading in the updateFilters$ effect maybe. I think we shouldn't spread here either though.

Reply:
I agree

Reply:
79f79835f55b69ee2cfde0cfdea1866e3572bd3f

Reply:
changed

Reply:
the `buildMetadataFromRequest` function seems to just overcomplicate something that should be pretty simple. It is also only used once. I'm ok with keeping it if you want, but I think it should be renamed to `buildCategoryPageMetadataFromRequest` if it is kept. Also, I don't think there is any reason to have an array of keys, and then to iterate over those keys. Why not just return:
```
return {
    id: request.id,
    applied_sort_option: request.applied_sort_option,
    applied_sort_direction: request.applied_sort_direction,
    current_page: request.current_page,
    page_size: request.page_size
}
```
If there is a purpose to the way you wrote it, please explain, because I'm not getting it.

Reply:
Because most of the request fields are optional, and you don't want to overwrite the metadata fields with undefined if they aren't present in the request

Reply:
that does make sense